src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java

```diff
@@ -204,10 +204,29 @@ protected Object getValueInternal(ProgressListener listener) throws Exception {
             // actually parse the data
             if (input != null) {
                 CancellingInputStream is = new CancellingInputStream(input, listener);
-                return complexPPIO.decode(is);
+                Object result = complexPPIO.decode(is);
+                if (result == null || complexPPIO.getType().isInstance(result)) {
+                    return result;
+                }
+                // Some text parsers return a Map when it can not be converted to
+                // the proper type.  Detect those errors here rather than later.
+                throw new IllegalArgumentException(
+                        "Decoded result is not a "
+                                + complexPPIO.getType().getName()
+                                + ", got a: "
+                                + result);
```

Comment:
Is this trying to display `result` or should it be showing the type of `result`?

Reply:
I changed it to show the type of result.